src/huggingface_hub/hf_api.py

```diff
@@ -1194,6 +1195,197 @@ def list_spaces(
+        response = requests.delete(
+            url=f"{self.endpoint}/api/{repo_type}s/{repo_id}/like",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(response)
+
+    @validate_hf_hub_args
+    def list_liked_repos(
+        self,
+        user: Optional[str] = None,
+        *,
+        token: Optional[str] = None,
+    ) -> UserLikes:
+        """
+        List all repos liked by a user on huggingface.co.
+
+        This list is public so token is optional. If `user` is not passed, it defaults to
+        the logged in user.
+
+        See also [`like`] and [`unlike`].
+
+        Args:
+            user (`str`, *optional*):
+                Name of the user for which you want to fetch the likes.
+            token (`str`, *optional*):
+                A valid authentication token (see https://huggingface.co/settings/token).
+                Used only if `user` is not passed to implicitly determine the current
+                user name.
+
+        Returns:
+            [`UserLikes`]: object containing the user name, the total count of likes and
+            3 lists of repo ids (1 for models, 1 for datasets and 1 for Spaces).
+
+        Raises:
+            [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+                If `user` is not passed and no token found (either from argument or from machine).
+
+        Example:
+        ```python
+        >>> from huggingface_hub import list_liked_repos
+
+        >>> likes = list_liked_repos("julien-c")
+
+        >>> likes.user
+        "julien-c"
+
+        >>> likes.total
+        626
+
+        >>> likes.models
+        ["osanseviero/streamlit_1.15", "Xhaheen/ChatGPT_HF", ...]
+        ```
+        """
+        # User is either provided explicitly or retrieved from current token.
+        if user is None:
+            me = self.whoami(token=token)
+            if me["type"] == "user":
+                user = me["name"]
+            else:
+                raise ValueError(
+                    "Cannot list liked repos. You must provide a 'user' as input or be"
+                    " logged in as a user."
+                )
+
+        path = f"{self.endpoint}/api/users/{user}/likes"
+        headers = self._build_hf_headers(token=token)
+
+        r = requests.get(path, headers=headers)
+        hf_raise_for_status(r)
+        data = r.json()
+        # Looping over a list of items similar to:
+        #   {
+        #       'createdAt': '2021-09-09T21:53:27.000Z',
+        #       'repo': {
+        #           'name': 'PaddlePaddle/PaddleOCR',
+        #           'type': 'space'
+        #        }
+        #   }
+        # Let's loop 3 times over the received list. Less efficient but more straightforward to read.
+        return UserLikes(
+            user=user,
+            total=data["totalCount"],
+            total_visible=len(data["visibleLikes"]),
```

Comment:
I find this structure/API slightly uninuitive. 

`total` will give you a number, but `len(models)` will give you a different number. Maybe `total` should be the visible ones, and then we could have a `total_private`? But I'm really not sure; no strong thoughts

Reply:
or just not expose either total? (and let users do `len(likes)`)

Reply:
I also feel it's weird to have 2 totals but I didn't come up with a "better" way. With the current implementation `UserLikes.total` is equal to the number we can now see on everyone's profile (for example [1106 for you](https://huggingface.co/datasets/open-source-metrics/hf-doc-google-analytics)) which I think would be the biggest source of confusion if it wasn't the case.

The `total_visible` is more there to remind users that what then can see is less that the real total (1092 in your case).

Reply:
i think we should just align on the hub side to only display "visible likes" cc @gary149 @coyotte508 @krampstudio what do you say?

Reply:
yes but again we will likely change this on the hub side anyways

Reply:
I removed the `total` and `total_visible` in https://github.com/huggingface/huggingface_hub/pull/1254/commits/bffb3a3ef4ec726501f4692ac3b672e71f2cf15d. At least we can merge this PR and potentially put back a property once this is figured out in https://github.com/huggingface/moon-landing/issues/4850 -if ever useful-